snowflake/datadog_checks/snowflake/check.py

```diff
@@ -209,8 +210,8 @@ def connect(self):
     @AgentCheck.metadata_entrypoint
     def _collect_version(self):
         try:
-            raw_version = self.execute_query_raw("select current_version();")
-            version = raw_version[0][0]
+            raw_version = next(self.execute_query_raw("select current_version();"))
+            version = raw_version[0]
```

Comment:
why is the second `[0]` not required here?

Reply:
Because of the `next` right above. When using `fetchall()`, `raw_version` was a list of tuples, and thus we were accessing the first element of the first item in the list. With `fetchone()`, `execute_query_raw` is returning an iterator, for which we're taking the first element already with `next`. Thus, in the new version of the code, `raw_version` is a tuple from which we're only interested in the first element.